test/test_ops.py

```diff
@@ -125,13 +131,146 @@ def test_inplace_gradgrad(self, device, dtype, op):
         self._gradgrad_test_helper(device, dtype, op, self._get_safe_inplace(op.get_inplace()))
 
 
-class TestOut(TestCase):
+class TestCommon(JitCommonTestCase):
     exact_dtype = True
 
+    # Compares variant's backward
+    # NOTE: verifies it fails when the forward fails
+    def check_variant_backward(self, input, forward_result, expected_grad, expected_exception):
+        variant_exception_during_backwards = False
+        try:
+            forward_result.sum().backward()
+            variant_grad = input.grad
+            input.grad = None
+        except Exception as e:
+            if not expected_exception:
+                self.fail("Unexpected exception during backwards!")
+            variant_exception_during_backwards = True
+
+        if expected_exception != variant_exception_during_backwards:
+            self.fail("Unexpected success during backwards!")
+
+        if not expected_exception:
+            self.assertEqual(variant_grad, expected_grad)
+
+    # Tests that the forward and backward passes of operations produce the
+    #   same values for the cross-product of op variants (function, method, inplace)
+    #   and runtimes (eager, traced, scripted).
+    # TODO WARNING: inplace x {traced, scripted} not currently tested
+    @ops(op_db)
+    def test_variant_consistency(self, device, dtype, op):
+        samples = op.sample_inputs(device, dtype, requires_grad=True)
+        if len(samples) == 0:
+            self.skipTest("Skipped! No sample inputs!")
+
+        for sample in samples:
+            # Computes expected forward
+            expected_forward = op(sample.input, *sample.args, **sample.kwargs)
+
+            # Computes expected backward
+            # NOTE: backward may fail for some dtypes
+            exception_during_backwards = False
+            expected_grad = None
+            try:
+                expected_forward.sum().backward()
+                expected_grad = sample.input.grad
+                sample.input.grad = None
+            except Exception as e:
+                exception_during_backwards = True
+
+            # Acquires variants to test
+            method = op.get_method()
+            inplace = op.get_inplace()
+            variants = (v for v in (method, inplace) if v is not None)
+
+            # Test eager consistency
```

Comment:
I'm not sure if we need to be testing eager's consistency with itself here since it's not JIT related